tests/common/ixia/ixia_fixtures.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+import pprint
+# from common.devices import SonicHost, Localhost, PTFHost, EosHost, FanoutHost
+from common.devices import FanoutHost
+""" 
+In an IXIA testbed, there is no PTF docker. 
+Hence, we use ptf_ip field to store IXIA API server. 
+This fixture returns the IP address of the IXIA API server.
+"""
+@pytest.fixture(scope = "module")
+def ixia_api_serv_ip(testbed):
+    return testbed['ptf_ip']
+
+"""
+Return the username of IXIA API server
+"""
+@pytest.fixture(scope = "module")
+def ixia_api_serv_user(duthost):
+    return duthost.host.options['variable_manager']._hostvars[duthost.hostname]['secret_group_vars']['ixia_api_server']['user']
+
+"""
+Return the password of IXIA API server
+"""
+@pytest.fixture(scope = "module")
+def ixia_api_serv_passwd(duthost):
+    return duthost.host.options['variable_manager']._hostvars[duthost.hostname]['secret_group_vars']['ixia_api_server']['password']
+
+"""
+Return REST port. 
+"""
+@pytest.fixture(scope = "module")
+def ixia_api_serv_port(duthost):
+    return duthost.host.options['variable_manager']._hostvars[duthost.hostname]['secret_group_vars']['ixia_api_server']['rest_port']
+
+"""
+IXIA PTF can spawn multiple session on the same REST port. Optional for LINUX, Rewuired for windows 
+Return the session ID. 
+"""
+@pytest.fixture(scope = "module")
+def ixia_api_serv_session_id(duthost):
+    return duthost.host.options['variable_manager']._hostvars[duthost.hostname]['secret_group_vars']['ixia_api_server']['session_id']
+  
+"""
+IXIA chassis are leaf fanout switches in the testbed.
+This fixture returns the hostnames and IP addresses of the IXIA chassis in the dictionary format.
+"""  
+@pytest.fixture(scope = "module")
+def ixia_dev(duthost, ansible_adhoc, conn_graph_facts, creds):
```

Comment:
as discussed in the meeting, this needs to be part of the fanouthosts fixture and abstracted in common/devices.py. Please look at the EosHost or OnyxHost for examples

Reply:
I have removed this fixture from ixia_fixtures.py, instead I am taking fanout_graph_facts as input and designed new class "IxiaFanoutManager" defined in ixia_helpers.py to handle and extract chassis IP address.